pkg/trustedresources/verifier/verifier.go

```diff
@@ -72,21 +73,28 @@ func FromPolicy(ctx context.Context, k8s kubernetes.Interface, policy *v1alpha1.
 		if err != nil {
 			return nil, fmt.Errorf("authority %q contains an invalid hash algorithm: %w", a.Name, err)
 		}
-		if a.Key.Data == "" && a.Key.SecretRef == nil {
+		if a.Key.Data == "" && a.Key.SecretRef == nil && a.Key.KMS == "" {
 			return nil, ErrorEmptyKey
 		}
-		if a.Key.Data != "" {
+		switch {
+		case a.Key.Data != "":
 			v, err := fromData([]byte(a.Key.Data), algorithm)
 			if err != nil {
 				return nil, fmt.Errorf("failed to get verifier from data: %w", err)
 			}
 			verifiers = append(verifiers, v)
-		} else if a.Key.SecretRef != nil {
+		case a.Key.SecretRef != nil:
 			v, err := fromSecret(ctx, fmt.Sprintf("%s%s/%s", keyReference, a.Key.SecretRef.Namespace, a.Key.SecretRef.Name), algorithm, k8s)
 			if err != nil {
 				return nil, fmt.Errorf("failed to get verifier from secret: %w", err)
 			}
 			verifiers = append(verifiers, v)
+		case a.Key.KMS != "":
+			v, err := kms.Get(ctx, a.Key.KMS, algorithm)
+			if err != nil {
+				return nil, fmt.Errorf("%w: %v", ErrorFailedLoadKMSKey, err)
```

Comment:
Ah yes. The reason is that the error from `kms.Get` has a private field, I cannot assert the error with `error.is`
https://github.com/sigstore/sigstore/blob/630a6277a539b55a854fe77049646ff4cad7c103/pkg/signature/kms/kms.go#L28-L31

Reply:
I think it's because we're using `%v` for the underlying error instead of `%w` (because `%w` is being used for `ErrorFailedLoadKMSKey`).

We'd need to implement a [wrapped error](https://go.dev/blog/go1.13-errors) ourselves for this to work, since fmt.Errorf is returning it's own wrapped error type rather than a specific type we want to use.

Reply:
I changed this part to wrap the error from `kms.Get`, figure out a way to test one error from it